specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Pipeline.json

```diff
@@ -1148,10 +1148,11 @@
     },
     "StoredProcedureParameterType": {
       "description": "Stored procedure parameter type.",
-      "type": "string",
+      "type": "object",
```

Comment:
If all allowed enum values are a string, why change to object?

Reply:
This is to support Expressions on these properties. OpenApi 2.0 doesn't support oneOf, and we are forced to use object type in order to support parameterization, which is an important feature of ADF for our customers. It allows them to treat their entities  as templates and pass different values for properties.